src/rustdoc_schema.graphql

```diff
@@ -429,6 +429,29 @@ interface Variant implements Item {
 
   # own edges
   field: [StructField!]
+
+  """
+  The discriminant, if explicitly specified.
+  """
+  discriminant: Discriminant
```

Comment:
I think "if explicitly specified" is going to cause some trouble with writing lints.

For example, how would this schema support a lint that catches the following breakage:
```rust
#[derive(PartialOrd)]
pub enum Example {
    // to break: swap these two lines
    A,
    B,
}
```
This is breaking in two ways: the implicit discriminant changes, and the enum sorts differently than before.

What do you think about defining the `discriminant` edge as `The discriminant, if one is defined either explicitly or implicitly.` + a link to the Rust reference and possibly a couple of inline examples.

A possible representation for the `VertexKind::Discriminant` variant might be something like:
```rust
pub enum DiscriminantValue<'a> {
    Explicit(&'a Discriminant),
    Implicit(Arc<str>),  // holds the `value`, and always has `expr = "_"`
}
```

Reply:
A possible other extension to the schema, assuming you agree with the above, might be to add an `explicit: Boolean!` property to `type Discriminant`, for the sake of better diagnostic messages.